src/include/codegen/barrier.h

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// barrier.h
+//
+// Identification: src/include/codegen/barrier.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <atomic>
+#include <boost/thread/barrier.hpp>
+
+#include "common/logger.h"
+#include "codegen/codegen.h"
+#include "codegen/utils/oa_hash_table.h"
+
+namespace peloton {
+namespace codegen {
+
+class Barrier
+{
+
+public:
```

Comment:
I suddenly found that the code style is not consistent with other files. Don't forget to re-format it when you have time later!